core/controllers/galleries.py

```diff
@@ -95,6 +95,37 @@ def get(self):
         })
         self.render_template('galleries/gallery.html')
 
+class GallerySearchPage(base.BaseHandler):
+    """The exploration gallery search page."""
+
+    PAGE_NAME_FOR_CSRF = 'gallery'
+
+    def get(self):
+        """Handles GET requests."""
+        gallery_handler = GalleryHandler(self.request, self.response)
+        gallery_handler.get_data()
+        explorations_list = gallery_handler.values['explorations_list']
+        query_response = self.request.get('q').split()
+        query_string_list = [x for x in query_response
```

Comment:
This is so that we can pass the search query to the front end so that it can be bound to `$scope.searchQuery` in the front (and the search query will show up in the search bar)